felix/deltatracker/delta_tracker.go

```diff
@@ -43,6 +43,16 @@ import (
 // keys that are in the dataplane map but not in the desired map.  The
 // pending maps are exposed via the IterPendingUpdates and IterPendingDeletions
 // methods.
+//
+// It's not recommended to use mutable objects in a DeltaTracker.  Updating a
+// mutable object while it is in the DeltaTracker will corrupt its internal
+// state.  By design, the DeltaTracker aliases the Desired and Dataplane values
+// if they happen to be equal. So, to safely mutate a value, you must
```

Comment:
I think you're saying that the DeltaTracker doesn't perform copies (either shallow or deep).  Is the more important point that it doesn't copy the data when initially passed to it, or that it doesn't copy data when moving between Desired and Dataplane?  (Or both equally.)